api/src/main/java/io/druid/data/input/impl/CSVParseSpec.java

```diff
@@ -80,23 +94,27 @@ public void verify(List<String> usedCols)
   @Override
   public Parser<String, Object> makeParser()
   {
-    return new CSVParser(Optional.fromNullable(listDelimiter), columns);
+    if (firstRowIsHeader) {
```

Comment:
This could just be `return new CSVParser(Optional.fromNullable(listDelimiter), columns, firstRowIsHeader)`